mail/api/account.py

```diff
@@ -60,53 +54,26 @@ def get_account_request(request_key: str) -> dict:
 def create_account(request_key: str, first_name: str, last_name: str, password: str) -> None:
 	"""Create a new user account"""
 
-	account_request, email, tenant, role = frappe.db.get_value(
-		"Mail Account Request", {"request_key": request_key}, ["name", "email", "tenant", "role"]
+	account_request, email, tenant, role, account = frappe.db.get_value(
+		"Mail Account Request", {"request_key": request_key}, ["name", "email", "tenant", "role", "account"]
 	)
 
+	user_email = account if account else email
+
 	user = frappe.new_doc("User")
 	user.first_name = first_name
 	user.last_name = last_name
-	user.email = email
-	user.owner = email
+	user.email = user_email
+	user.owner = user_email
 	user.new_password = password
-	user.append_roles(role)
+	user.append_roles("Mail User")
```

Comment:
Not sure if the same function is called for self-signed. self-signed users should not have the Mail User role. Mail User role is to create a Mail Account, which can't be created for a personal email.